src/invidious/frontend/pagination.cr

```diff
@@ -3,6 +3,30 @@ require "uri"
 module Invidious::Frontend::Pagination
   extend self
 
+  private def first_page(str : String::Builder, locale : String?, url : String, previous_page_button : Bool)
+    if previous_page_button
+      str << %(<noscript>)
+    end
```

Comment:
Why are you adding a `<noscript>` tag here? Mutating the "first-page" button to a "previous-page" button on the client side would be much simpler, and would reduce the complexity server side.

Reply:
Yes, but this code needed to have `<noscript>` on the server side, because it checked the `previous_page_button` preferences option.